test/unit/spec/narrative_core/kbaseNarrativeDownloadPanel-spec.js

```diff
@@ -83,30 +90,148 @@ define([
+                body: /run_job/,
+                response: [fakeJobId],
+            });
+            // instant finish to the job!
+            Mocks.mockJsonRpc1Call({
+                url: Config.url('execution_engine2'),
+                body: /check_job/,
+                response: {
+                    running: 12345,
+                    finished: 67890,
+                    status: 'completed',
+                    job_output: {
+                        result: [
+                            {
+                                shock_id: 'some_shock_id',
+                            },
+                        ],
+                    },
+                },
+            });
+            // some logs
+            Mocks.mockJsonRpc1Call({
+                url: Config.url('execution_engine2'),
+                body: /execution_engine2.get_job_logs/,
+                response: {
+                    count: 1,
+                    last_line_number: 1,
+                    lines: [
+                        {
+                            line: 'a bit of log',
+                            is_error: 0,
+                        },
+                    ],
+                },
+            });
+
+            const dlUrlRegex = new RegExp(Config.url('data_import_export'));
+            jasmine.Ajax.stubRequest(dlUrlRegex);
+
+            await initDownloadPanel(null, exporterCache);
+            const exportBtns = $div.find('.kb-data-list-btn');
+
+            await TestUtil.waitForElementState(
+                document,
+                () => {
+                    return document.getElementById('hiddenDownloader') !== null;
+                },
+                () => {
+                    exportBtns[0].click();
+                }
+            );
+            const iframe = document.getElementById('hiddenDownloader');
+            expect(iframe).not.toBeNull();
+            const expectedUrl =
+                'https://ci.kbase.us/services/data_import_export/download?id=some_shock_id&del=1&name=fake_test_object.SOME_FORMAT.zip&url=https%3A%2F%2Fci.kbase.us%2Fservices%2Fshock-api';
+            expect(iframe.src).toBe(expectedUrl);
+        });
+
+        it('Should create an app cell with the staging exporter', async () => {
+            /* use mock for Jupyter.narrative.addAndPopulateApp */
+            spyOn(Jupyter.narrative, 'addAndPopulateApp');
+            const exporterCache = {
+                lastUpdateTime: 100,
+                types: {
+                    [objType]: {
+                        export_functions: {
+                            STAGING: 'gets_overwritten_right_now',
+                        },
+                    },
+                },
+            };
+            await initDownloadPanel(null, exporterCache);
+            const exportBtns = $div.find('.kb-data-list-btn');
+            // expect 2 buttons - STAGING, JSON, in that order.
+            expect(exportBtns.length).toEqual(2);
```

Comment:
worth checking the button text?

Reply:
That's done in an earlier test. I figured it was redundant.